I2C.bme280.oled/simple_remote.py

```diff
@@ -0,0 +1,159 @@
+    raw_html = urllib.request.urlopen(url).read().decode('utf-8').rstrip()
+    return raw_html
+
+
+def get_remote_humid_and_temp(id):
+    forecast_url = yanpiws_ajax_url + 'humidity&id=' + str(id)
+    return json.loads(get_string_from_url(forecast_url))
+
+
+def get_remote_forecast():
+    forecast_url = yanpiws_ajax_url + 'forecast_full_json'
+    return json.loads(get_string_from_url(forecast_url))
+
+
+def get_remote_sun():
+    data = json.loads(get_string_from_url(yanpiws_ajax_url + "sunrise"))
+    data.update(json.loads(get_string_from_url(yanpiws_ajax_url + "sunset")))
+    return "☀" + str(data['sunrise']) + " ○" + str(data['sunset'])
+
+
+def show_info(wait):
+
+    # fetch the cooked up json -> strings
+    forecast = {}
+    humid_and_temp1 = {}
+    try:
+        no_error = True
+        humid_and_temp1 = get_remote_humid_and_temp(yanpiws_temp_1)
+        forecast = get_remote_forecast()
+        first_line = get_remote_sun()
+    except:
+        first_line = 'Error - check logs :('
+        no_error = False
+
+    second_line = ''
+    third_line = ''
+
+    if no_error and humid_and_temp1[0]['temp'] != 'NA':
+        second_line = str(int(float(humid_and_temp1[0]['temp']))) + '°' + humid_and_temp1[0]['label']
+
+    if no_error  and  forecast[0]:
+        third_line = str(int(float(forecast[0]['temperatureMax']))) + '°' + forecast[0]['icon']
+
+    full_path = os.path.dirname(os.path.abspath(__file__)) + "/"
+    font2 = ImageFont.truetype(full_path + "Lato-Heavy.ttf", 14)
+    font1 = ImageFont.truetype(full_path + "Lato-Heavy.ttf", 20)
+
+    with canvas(device) as draw:
+        draw.text((0, 0), first_line, font=font2, fill="white")
+        draw.text((0, 17), second_line, font=font1, fill="white")
+        draw.text((0, 41), third_line, font=font1, fill="white")
+
+    my_logger.debug(f"Weathercaster: simple Updated screen, posted {last_seen_temp} temp. Waiting {wait} seconds to update again.")
+
+
+def full_stack():
+    import traceback, sys
+    exc = sys.exc_info()[0]
+    stack = traceback.extract_stack()[:-1]  # last one would be full_stack()
+    if exc is not None:  # i.e. an exception is present
+        del stack[-1]  # remove call of full_stack, the printed exception
+        # will contain the caught exception caller instead
+    trc = 'Traceback (most recent call last):\n'
+    stackstr = trc + ''.join(traceback.format_list(stack))
+    if exc is not None:
+        stackstr += '  ' + traceback.format_exc().lstrip(trc)
+    return stackstr
+
+
+def main(device):
+    while True:
+        wait = 120
+        post_to_yanpiws()
+        show_info(wait)
+        time.sleep(wait)
+
+
+if __name__ == "__main__":
+    try:
+        my_logger.debug('Weathercaster: simple Starting ')
+        serial = i2c(port=args.bus, address=0x3C)
+        device = ssd1306(serial)
+        main(device)
+
+
+    except KeyboardInterrupt:
+        my_logger.debug("Weathercaster: simple Stopping(Ctrl + C) ")
+        pass
```

Comment:
`pass` is a no-op meant to fill a code block that would otherwise would be empty; it has no effect here.